Transcribe2.java

```diff
@@ -0,0 +1,264 @@
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.CopyObjectRequest;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectInputStream;
+import com.amazonaws.services.transcribe.AmazonTranscribe;
+import com.amazonaws.services.transcribe.AmazonTranscribeClient;
+import com.amazonaws.services.transcribe.AmazonTranscribeClientBuilder;
+import com.amazonaws.services.transcribe.model.DeleteTranscriptionJobRequest;
+import com.amazonaws.services.transcribe.model.GetTranscriptionJobRequest;
+import com.amazonaws.services.transcribe.model.GetTranscriptionJobResult;
+import com.amazonaws.services.transcribe.model.LanguageCode;
+import com.amazonaws.services.transcribe.model.Media;
+import com.amazonaws.services.transcribe.model.StartTranscriptionJobRequest;
+import com.amazonaws.services.transcribe.model.StartTranscriptionJobResult;
+import com.amazonaws.services.transcribe.model.TranscriptionJob;
+import com.amazonaws.services.transcribe.model.TranscriptionJobStatus;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class Transcribe2{
+
+	//@Autowired
+	//private Environment env;
+
+	private String bucketName = Constants.bucketName;
+	private Regions regions = Constants.awsSeerviceRegion;
+
+	public AmazonTranscribe transcribeClient() {
+		System.out.println("Intialize Transcribe Client");
+		AmazonTranscribe amazonTranscribe = AmazonTranscribeClientBuilder
+				.standard()
+				.withRegion(regions)
+				.build();
+
+		return amazonTranscribe;
+	}
+
+
+	public AmazonS3 s3Client() {
+		System.out.println("Intialize AWS S3 Client");
+		AmazonS3 s3client = AmazonS3ClientBuilder
+				.standard()
+				.withRegion(regions)
+				.build();
+
+		return s3client;
+	}
+
+	public void deleteFileFromAwsBucket(String fileName) {
+		System.out.println("Delete File from AWS Bucket "+fileName);
+		String key = fileName.replaceAll(" ", "_").toLowerCase();
+		//String newKey = key + "-" + System.currentTimeMillis();
+		//s3Client().copyObject(bucketName, key, bucketName, newKey);
+		try {
+			Thread.sleep(1000);
+			s3Client().deleteObject(bucketName, key);
+		}catch(Exception e) {
+			e.printStackTrace();
+		}
+	}
+
+	public void testFileFromAwsBucket(String fileName) {
+		System.out.println("Delete File from AWS Bucket "+fileName);
+		String key = fileName.replaceAll(" ", "_").toLowerCase();
+		//String newKey = key + "-" + System.currentTimeMillis();
+		//s3Client().copyObject(bucketName, key, bucketName, newKey);
+		S3Object s3object = null;
+		S3ObjectInputStream inputStream = null;
+		try {
+			Thread.sleep(1000);
+			s3object = s3Client().getObject(bucketName, key);
+			inputStream = s3object.getObjectContent();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *inputStream*. Execution paths that do not contain closure statements were detected. Ensure *inputStream* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).